workflow-templates/cf-gcp-terraform.yml

```diff
@@ -73,11 +60,15 @@ jobs:
       - name: Setup Terraform
         uses: hashicorp/setup-terraform@v1
 
-      - name: Setup GCP Credentials
-        uses: google-github-actions/setup-gcloud@master
+      # In next step in case you're using CF terraform-google-cf-projectcfg module to configure workload identity federation change 123456789
+      # to your actual project's number. As for the service account - change it accordingly to a service account you use to run terraform code
+      # Note: if you use CF terraform-google-cf-projectcfg module and it's bootstrap script,
+      # your SA will look like 'terraform-iac-pipeline@{gcp_project_id}.iam.gserviceaccount.com'
+      - name: 'Authenticate to Google Cloud'
+        uses: 'google-github-actions/auth@v0'
         with:
-          service_account_key: ${{ secrets[matrix.secret_name] }}
-          export_default_credentials: true     
+          workload_identity_provider: 'projects/123456789/locations/global/workloadIdentityPools/github-actions/providers/github'
+          service_account: 'my-service-account@my-project.iam.gserviceaccount.com'
```

Comment:
From my understanding this needs to use `${{ matrix.<something> }}`, otherwise stages will not work.

Reply:
it was done, please check it again